ssg/utils.py

```diff
@@ -342,13 +351,9 @@ def apply_formatting_on_dict_values(source_dict, string_dict, ignored_keys=[]):
     """
     new_dict = {}
     for k, v in source_dict.items():
-        if k not in ignored_keys:
-            if isinstance(v, dict):
-                new_dict[k] = apply_formatting_on_dict_values(v, string_dict, ignored_keys)
-            elif isinstance(v, str):
-                new_dict[k] = v.format(**string_dict)
-            else:
-                new_dict[k] = v
+        if ignored_keys is None or k not in ignored_keys:
```

Comment:
It's better to make `ignored_keys` an empty list/set if it is not given, and to do so at the top of the function. That way, the `if` statement may be made human-readable.